### PR TITLE
Moving to fastapi 

### PR DESCRIPTION
**Pull request by Iddoyadlin**
wdyt?

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,83 +1,56 @@
...
+                app.state.usage[ip] = 0
+    app.state.usage[key] += 1
+    return app.state.usage[key] > app.state.limit
+
+
+@app.middleware("http")
+async def is_limited(request: Request, call_next):
+    ip_address = request.client.host
+    if request_is_limited(key=ip_address):
+        raise HTTPException(status_code=429, detail="Limited!")
```
Comment (ishefi) [inline app.py:54]: right now this returns 500, I think it's due to the fact it's in the middlewhere and not the handler itself. Take a look at: https://stackoverflow.com/questions/61358669/raise-exception-in-python-fastapi-middleware

File: templates/faq.html [hunk 2]
```diff
@@ -22,7 +23,7 @@
 
 <header>
 <h1>שאלות נפוצות</h1>
-<div w3-include-html="menu.html"></div>
+<div w3-include-html="/templates/menu.html"></div>
```
Comment (ishefi) [inline templates/faq.html:26]: I think it should be `static/menu.html` (this is how you did it elsewhere)

File: templates/index.html [hunk 3]
```diff
@@ -88,7 +89,11 @@ <h1>סמנטעל</h1>
 <footer>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
     <script src="https://cdn.statically.io/gh/zenorocha/clipboard.js/v2.0.10/dist/clipboard.min.js"></script>
-    <script src="js/semantle.js?version={{js_version}}"></script>
+    <script src="static/semantle.js?version={{js_version}}"></script>
+
+    <script src="//static/semantle.js?version={{js_version}}"></script>
```
Comment (ishefi) [inline templates/index.html:94]: this is wrong, leave only the first `script` (here and elsewhere)

File: templates/faq.html [hunk 4]
```diff
@@ -98,7 +99,7 @@ <h1>שאלות נפוצות</h1>
 </body>
 <footer>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
-    <script src="js/semantle.js?version={{js_version}}"></script>
+    <script src="//static/js/semantle.js?version={{js_version}}"></script>
```
Comment (ishefi) [inline templates/faq.html:102]: this doesn't work - here and elsewhere

**Review by ishefi — CHANGES_REQUESTED** [review]
A change to the Procfile is needed too, we should probably use gunicorn

File: app.py (hunk 1 above)
Comment (Iddoyadlin) [reply app.py]: 032dc228c2ccc76bd24ee0ec5ef107262a8c0c9e

Comment (Iddoyadlin) [reply app.py]: image

File: templates/faq.html (hunk 2 above)
Comment (Iddoyadlin) [reply templates/faq.html]: cd2e2fce690a55f3b5580846cc97d2fd385fe902

File: templates/faq.html (hunk 4 above)
Comment (Iddoyadlin) [reply templates/faq.html]: c87f89be6451ce4e7e5f810b019adb7b38badd2f

Comment (Iddoyadlin) [reply templates/faq.html]: without this, when going to 127.0.0.1:8000/faq, I get a request for 127.0.0.1:8000/faq/static/semantle.js?version={{js_version}}

File: templates/index.html (hunk 3 above)
Comment (Iddoyadlin) [reply templates/index.html]: https://github.com/ishefi/semantle-he/commit/c87f89be6451ce4e7e5f810b019adb7b38badd2f

**Comment by Iddoyadlin** [comment]
> A change to the Procfile is needed too, we should probably use gunicorn

can you explain? you know this better than me

**Comment by Iddoyadlin** [comment]
15a716fe22b5c5158feed2ab8c5ffe536471d4c5
@ishefi added random string to api/distance query to try and avoid caching weirdness on iphone

**Pull request merged**